### PR TITLE
HPCC-19366 Add abilty to recreate query

### PR DESCRIPTION
**Pull request by miguelvazq**
Add WsWorkunits/WURecreateQuery capabilities to ECL Watch.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review

**Comment by miguelvazq** [comment]
FYI @afishbeck

File: esp/src/eclwatch/nls/hpcc.js [hunk 1]
```diff
@@ -751,6 +756,7 @@ define({root:
     Unsuspended: "Unsuspended",
     Up: "Up",
     UpdateCloneFrom: "Update Clone From",
+    UpdateDFs: "Update DFs",
```
Comment (afishbeck) [inline esp/src/eclwatch/nls/hpcc.js]: Should be capital S... DFS.

**Review by afishbeck — COMMENTED** [review]
Actually there is another parameter you need to set... req->setRepublish(!optNoPublish);

You can add another boolean checkbox to the form, but with opposite logic.  "Do not republish".  It should default to false which means "YES republish) and the boolean value passed to "setRepublish" should be the opposite of the value for "Do not republish".

It just makes the choice easier.  We default to YES, but change to NO if they check the box.

**Comment by miguelvazq** [comment]
Comments made as per @afishbeck comments.

**Review by afishbeck — APPROVED** [review]
@miguelvazq looks good.

**Comment by miguelvazq** [comment]
Remind me @afishbeck on === 0?

File: esp/src/eclwatch/QuerySetQueryWidget.js [hunk 2]
```diff
@@ -475,18 +498,64 @@ define([
...
+
+        _onRecreateQueries: function () {
+            if (this.recreateForm.validate()) {
+                var context = this;
+                var success = false;
+                arrayUtil.forEach(this.recreateQueriesGrid.store.data, function (item, idx) {
+                        var request = domForm.toObject(context.id + "RecreateForm");
+                        request.Republish === "on" ? request.Republish = 0 : request.Republish = 1;
+                        request.AllowForeignFiles = "on" ? request.AllowForeignFiles = 0 : request.AllowForeignFiles = 1;
+                        request.UpdateDfs = "on" ? request.UpdateDfs = 0 : request.UpdateDfs = 1;
```
Comment (afishbeck) [inline esp/src/eclwatch/QuerySetQueryWidget.js]: Should some of these be "==" or "===", rather than "="?

**Comment by afishbeck** [comment]
Not sure what you mean about the on === 0 question, can you explain?

But that lead me to find another potential issue, which I commented on.

File: esp/src/eclwatch/QuerySetQueryWidget.js (hunk 2 above)
Comment (miguelvazq) [reply esp/src/eclwatch/QuerySetQueryWidget.js]: Yes, the first one comparison in the ternary should be I will update.

**Comment by miguelvazq** [comment]
Meaning is 0 === true / on?

**Comment by afishbeck** [comment]
For the republish 1 is true / on. but I wanted to reverse the logic for EclWatch users.  Which is why I suggested implementing the UI part as "Do not publish" and reversing the value when setting "republish".

**Comment by miguelvazq** [comment]
@afishbeck Should be updated appropriately now.

**Comment by miguelvazq** [comment]
In that case I re-wrote the ternary to make it clearer.

**Review by afishbeck — APPROVED** [review]
@miguelvazq yes, better, thanks.  Looks good.

**Comment by miguelvazq** [comment]
Bump @GordonSmith

**Review by GordonSmith — APPROVED** [review]
Looks good.

**Comment by miguelvazq** [comment]
@richardkchapman Ready for merge.

**Pull request merged**